envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt

```diff
@@ -234,29 +289,20 @@ data class Incoming(
 }
 
 data class Outgoing(
-    val dependencies: List<Dependency> = emptyList()
+    val serviceDependencies: List<ServiceDependency> = emptyList(),
+    val domainDependencies: List<DomainDependency> = emptyList(),
+    val allServicesDependencies: Boolean = false,
+    val defaultServiceSettings: DependencySettings = DependencySettings()
 ) {
-    fun containsDependencyForService(service: String) = serviceDependencies.containsKey(service)
-
-    // not declared in primary constructor to exclude from equals(), copy(), etc.
-    private val domainDependencies: Map<String, DomainDependency> = dependencies
-        .filterIsInstance<DomainDependency>()
-        .map { it.domain to it }
-        .toMap()
-
-    private val serviceDependencies: Map<String, ServiceDependency> = dependencies
-        .filterIsInstance<ServiceDependency>()
-        .map { it.service to it }
-        .toMap()
-
-    fun getDomainDependencies(): Collection<DomainDependency> = domainDependencies.values
-
-    fun getServiceDependencies(): Collection<ServiceDependency> = serviceDependencies.values
-
     data class TimeoutPolicy(
-        val idleTimeout: Duration?,
-        val requestTimeout: Duration?
-    )
+        val idleTimeout: Duration = DEFAULT_IDLE_TIMEOUT,
```

Comment:
Why do we need this default here? `default` is configurable by `properties.egress.commonHttp.idleTimeout` and value is defined in [CommonHttpProperties](https://github.com/allegro/envoy-control/blob/master/envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotProperties.kt#L224)
This seems confusing and error-prone.

Reply:
There are classes (ProxySettings,Outgoing,Incoming itp) with default contractor for test purposes. I decided to use nonNull fields in this class and those default are used only in tests. I agree with you It can be confusing. I change it to nullable fields